Bring up adaptive navigation for reply study

Renzo-Olivares opened the pull request:
[Material spec](https://material.io/design/material-studies/reply.html#product-architecture) outlines the navigation experience for reply as follows:
Desktop: Navigation Drawer
Tablet: Navigation Rail
Mobile: BottomAppBar

This PR introduces a [ResponsiveWidget](https://medium.com/flutter-community/making-cross-platform-flutter-landing-page-responsive-7fffe0655970) class that uses a LayoutBuilder to return the appropriate navigation layout based on screen size given by MediaQueries. 

![adaptive-nav](https://user-images.githubusercontent.com/948037/88444226-fc6b3f00-cdd0-11ea-879d-b24a8f134e8e.gif)

Renzo-Olivares committed:
Bring up adaptive navigation for reply study, change between nav rail and bottom app bar

clocksmith commented on lib/studies/reply/inbox.dart:
```diff
@@ -1,12 +1,166 @@
+    );
+  }
+
+  void _onItemTapped(int index) {
```
_onItemSelected works better with a mix of taps and clicks

clocksmith replied:
maybe onDestinationSelected?

Renzo-Olivares replied:
I'll try that

clocksmith commented on lib/studies/reply/responsive_widget.dart:
```diff
@@ -0,0 +1,39 @@
+      {Key key, this.desktopScreen, this.tabletScreen, this.mobileScreen})
+      : super(key: key);
+
+  static bool isDesktop(BuildContext context) {
```
Make a TODO note here to re-use the code in adaptive.dart once its finalized

rami-a commented on lib/studies/reply/inbox.dart:
```diff
@@ -1,12 +1,166 @@
+  @override
+  void didUpdateWidget(_BuildDesktopNav oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.extended != widget.extended) _isExtended = widget.extended;
```
Nit: I prefer: 

```suggestion
    if (oldWidget.extended != widget.extended) {
      _isExtended = widget.extended;
    }
```

goderbauer committed:
Bump animations to 1.1.2 (#231)

clocksmith committed:
Bump version for motion demos (#230)

* bump version for motion demos

Renzo-Olivares committed:
Shared y axis reverse fix (#236)

* Add reverse property to PageTransitionSwitcher that dictates whether the new widget transitions under or over the old child

* Remove duration that was used in testing

* update code segments

pedromassango committed:
Migrate Form autovalidate API (#245)

johnsonmh committed:
Ignore deprecated use of BottomNavigationBarItem.title  (#247)

* add ignore deprecated member use comments

* update code viewer

justinmc committed:
Fix and simplify transformation demo to ameliorate web issues (#246)

A few different fixes and tweaks to make the transformation demo more solid on web.

johnsonmh committed:
Ignore deprecated use of BottomNavigationBarItem.title (#248)

* add comment and update code segments

* format

* update code segments again

justinmc committed:
Version bump for IV updates (#249)

clocksmith committed:
Update README.md (#232)

Fix error in command

Renzo-Olivares committed:
Merge branch 'master' of github.com:flutter/gallery into reply-navigation-bringup

Renzo-Olivares committed:
Use brackets with if statements

Renzo-Olivares committed:
Add todo to refactor and use adaptive.dart when finalized

Renzo-Olivares committed:
Add comment about linkedhashmap

Renzo-Olivares committed:
reformat with dartfmt

Renzo-Olivares committed:
Rename onItemTapped -> onDestinationSelected

Renzo-Olivares committed:
reformat with dartfmt
